protocol-units/da-sequencer/node/src/storage/mod.rs

```diff
@@ -188,7 +191,84 @@ impl DaSequencerStorage for Storage {
 	}
 
 	fn produce_next_block(&self) -> Result<Option<SequencerBlock>, DaSequencerError> {
-		todo!()
+		let cf_pending = self.db.cf_handle(cf::PENDING_TRANSACTIONS).ok_or_else(|| {
+			DaSequencerError::StorageAccess("Missing colomn family: pending transactions".into())
+		})?;
+
+		// Collect and deserialize all pending transactions
+		let mut transactions: Vec<Transaction> = Vec::new();
+
+		let iter = self.db.iterator_cf(&cf_pending, rocksdb::IteratorMode::Start);
+		for item in iter {
+			let (_key, value) = item.map_err(|e| DaSequencerError::RocksDbError(e.to_string()))?;
+			let tx: Transaction = bcs::from_bytes(&value)
```

Comment:
You should keep the size of the Tx to help to evaluate the size of the block.

Reply:
I get the size directly here now, no deserialization occurs 
```rust
for item in iter {
	let (key, value) = item.map_err(|e| DaSequencerError::RocksDbError(e.to_string()))?;
	let tx_size = value.len() as u64;
```